packages/payment-processor/src/payment/erc20.ts

```diff
@@ -101,19 +212,32 @@ export function encodeApproveErc20(
     throw new Error('No payment network Id');
   }
   validateRequest(request, paymentNetworkId);
-  const signer = getSigner(signerOrProvider);
+  return encodeApproveAnyErc20(
+    request.currencyInfo.value,
+    getProxyAddress(request),
+    getSigner(signerOrProvider)
+  );
+}
 
-  const proxyAddress = getProxyAddress(request);
-  const tokenAddress = request.currencyInfo.value;
-  const erc20interface = ERC20Contract.connect(tokenAddress, signer).interface;
-  const encodedApproveCall = erc20interface.functions.approve.encode([
-    proxyAddress,
+/**
+ * Encodes the approval call to approve any erc20 token to be spent, with no limit.
+ * @param tokenAddress the ERC20 token address to approve
+ * @param spenderAddress the address granted the approval
+ * @param signerOrProvider the signer who owns ERC20 tokens
+ */
+export function encodeApproveAnyErc20(
```

Comment:
I'm not a fan of the name.
It encodes the approve method for the token in the parameter, not for any token. This sounds to me that I could use this to approve all tokens, or that it would give me something generic I could use against any token.

Reply:
Again, I tried my best to work around all the previous asumptions that we pay in request currency. For example `encodeApproveErc20` should be renamed `encodeApproveRequestErc20`or `encodeApproveDefaultErc20` or something like that.

What do you suggest? I wanted to avoid breaking changes.